ui-search/src/main/java/app/tivi/home/search/SearchFragment.kt

```diff
@@ -18,97 +18,71 @@ package app.tivi.home.search
 import android.view.ViewGroup
-import androidx.appcompat.widget.SearchView
+import android.view.ViewGroup.LayoutParams.MATCH_PARENT
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.ui.platform.ComposeView
 import androidx.core.net.toUri
-import androidx.core.view.marginBottom
-import androidx.core.view.updatePadding
+import androidx.fragment.app.Fragment
 import androidx.fragment.app.viewModels
+import androidx.lifecycle.lifecycleScope
 import androidx.navigation.fragment.findNavController
-import app.tivi.FragmentWithBinding
-import app.tivi.data.entities.TiviShow
-import app.tivi.extensions.doOnLayouts
-import app.tivi.extensions.hideSoftInput
-import app.tivi.extensions.toActivityNavigatorExtras
-import app.tivi.home.search.databinding.FragmentSearchBinding
-import app.tivi.ui.createSharedElementHelperForItemId
-import app.tivi.ui.recyclerview.HideImeOnScrollListener
-import app.tivi.ui.transitions.GridToGridTransitioner
+import app.tivi.common.compose.LogCompositions
+import app.tivi.common.compose.ProvideDisplayInsets
+import com.google.android.material.composethemeadapter.MdcTheme
 import dagger.hilt.android.AndroidEntryPoint
-import javax.inject.Inject
+import kotlinx.coroutines.channels.Channel
 
 @AndroidEntryPoint
-internal class SearchFragment : FragmentWithBinding<FragmentSearchBinding>() {
+internal class SearchFragment : Fragment() {
     private val viewModel: SearchViewModel by viewModels()
-
-    @Inject internal lateinit var controller: SearchEpoxyController
+    private val pendingActions = Channel<SearchAction>(Channel.BUFFERED)
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        GridToGridTransitioner.setupFirstFragment(this)
+
+        lifecycleScope.launchWhenStarted {
+            for (action in pendingActions) {
+                when (action) {
+                    is SearchAction.OpenShowDetails -> {
+                        findNavController().navigate("app.tivi://show/${action.showId}".toUri())
+                    }
+                    else -> viewModel.submitAction(action)
+                }
+            }
+        }
     }
 
-    override fun createBinding(
+    override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): FragmentSearchBinding {
-        return FragmentSearchBinding.inflate(inflater, container, false)
-    }
+    ): View? = ComposeView(requireContext()).apply {
+        layoutParams = ViewGroup.LayoutParams(MATCH_PARENT, MATCH_PARENT)
 
-    override fun onViewCreated(binding: FragmentSearchBinding, savedInstanceState: Bundle?) {
-        binding.searchRecyclerview.apply {
-            setController(controller)
-            addOnScrollListener(HideImeOnScrollListener())
-        }
+        setContent {
+            MdcTheme {
+                LogCompositions("MdcTheme")
```

Comment:
Do you want to keep these logs?

Reply:
For now yeah, I'll go through and take a pass through soon and remove them all